packages/ember-application/lib/system/application.js

```diff
@@ -772,6 +772,10 @@ var Application = Namespace.extend({
     this._bootPromise = null;
     this._bootResolver = null;
 
+    if (_loaded.application === this) {
```

Comment:
Is it worth putting a comment re: `willDestroy` - to prevent leaks the reference to the loaded application must be broken